### PR TITLE
[Merged by Bors] - Add examples for Transforms

### LINKED ISSUE #2400: Add a transform and Quaternion example section.
## How can Bevy's documentation be improved?
Currently there is not really a specific place to add transformation examples other than 2d or 3d. I think this would encourage more examples to be created about this topic. 

Examples that could be included:

- translation of object
- rotation of objects
- scaling of objects
-  translation, rotation, and scaling at the same time
- show the difference between global transform and Transform
- the 3d parenting example could be moved into here.
- more examples showcasing the various methods on Transform and Quaternion.

### PR DESCRIPTION
**Pull request by l4desu-mizu**
# Add Transform Examples

- Adding examples for moving/rotating entities (with its own section) to resolve #2400 

I've stumbled upon this project and been fiddling around a little. Saw the issue and thought I might just add some examples for the proposed transformations.
Mind to check if I got the gist correctly and suggest anything I can improve?

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
bors try

File: examples/transforms/global_vs_local_translation.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup.system())
+        .add_system(move_cubes.system())
+        .add_system(stop_too_far_local_distance.system())
+        .add_system(stop_too_far_global_distance.system())
+        .run();
+}
+
+fn stop_too_far_global_distance(
+    mut cubes: Query<(&GlobalTransform, &mut SomeEntity, &GlobalStop)>,
```
Comment (NathanSWard) [inline examples/transforms/global_vs_local_translation.rs:32]: If `GlobalStop` is not being read from, you should use a query filter.
```rust
Query<(&GlobalTransform, &mut SomeEntity), With<GlobalStop>)
```

Comment (NathanSWard) [reply examples/transforms/global_vs_local_translation.rs:32]: Ditto with `LocalStop` below.

File: examples/transforms/transform.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    mut transformabels: QuerySet<(
+        Query<(&Transform, &CubeState)>,
+        Query<(&mut Transform, &Center)>,
+    )>,
+) {
+    let distance: f32 = transformabels
+        .q0()
+        .iter()
+        .map(|(transform, cube)| (cube.start_pos - transform.translation).length())
+        .fold(0.0, |x, y| x + y);
```
Comment (NathanSWard) [inline examples/transforms/transform.rs:52]: you can use `.sum()` instead of a manual `fold`

**Review by NathanSWard — CHANGES_REQUESTED** [review]
This looks like a nice addition. 

Note, you'll also have to update the `README.md` inside the `examples/` directory.

**Comment by NathanSWard** [comment]
Also as a quick note, it's nice if inside the examples themselves, there are more comments explaining what's happening. 
Adding some comments to explain what each system does, what you're spawning in the `setup` systems, would be a nice addition :smile:

File: examples/transforms/rotation.rs [hunk 3]
```diff
@@ -0,0 +1,52 @@
+use bevy::prelude::*;
+
+use std::f32::consts::PI;
+
+const FULL_TURN: f32 = 2.0 * PI;
```
Comment (NathanSWard) [inline examples/transforms/rotation.rs:5]: You can use `TAU` instead of `2.0 * PI`

**Review by NathanSWard — CHANGES_REQUESTED** [review]

File: examples/transforms/rotation.rs (hunk 3 above)
Comment (alice-i-cecile) [reply examples/transforms/rotation.rs:5]: I laugh, but I don't hate the idea :p

**Comment by alice-i-cecile** [comment]
As the CI noted, you need to add these to the examples README :)

File: examples/transforms/global_vs_local_translation.rs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        ..Default::default()
+    });
+
+    commands.spawn_bundle(PointLightBundle {
+        transform: Transform::from_translation(Vec3::ONE * 3.0),
+        ..Default::default()
+    });
+
+    let main_entity_spawn: Transform = Transform::from_translation(Vec3::ZERO);
+    let subcomponent_transform = Transform::from_translation(Vec3::Y);
```
Comment (alice-i-cecile) [inline examples/transforms/global_vs_local_translation.rs:77]: I'd like some comments here explaining the setup in plain English, since it's rather involved.

File: examples/transforms/global_vs_local_translation.rs [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+fn move_cubes(mut cubes: Query<(&mut Transform, &SomeEntity)>, timer: Res<Time>) {
+    for (mut transform, _) in cubes.iter_mut().filter(|(_, cube)| !cube.stopped) {
+        // using the local transform we can move an entity along it's local z axis
+        let dir = transform.local_x();
+        transform.translation += dir * timer.delta_seconds();
+    }
+}
+
+fn setup(
```
Comment (alice-i-cecile) [inline examples/transforms/global_vs_local_translation.rs:61]: IMO you should try to mimic the order of your AppBuilder and the functions in your code, so you can easily find them.

File: examples/transforms/scale.rs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup.system())
+        .add_system(scale_cube.system())
+        .run();
+}
+
+fn scale_cube(mut cubes: Query<(&mut Transform, &mut Scaling)>, timer: Res<Time>) {
+    for (mut transform, mut cube) in cubes.iter_mut() {
+        if transform.scale.length() >= 10.0 {
+            cube.scale_axis *= -1.0;
```
Comment (alice-i-cecile) [inline examples/transforms/scale.rs:18]: This should be commented to explain what it does.

File: examples/transforms/transform.rs [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        let full_turn = transform
+            .looking_at(Vec3::ZERO, transform.local_y())
+            .rotation;
+        let incremental_turn_weight = cube.turn_speed * timer.delta_seconds();
+        transform.rotation = transform.rotation.lerp(full_turn, incremental_turn_weight);
+    }
+}
+
+fn scale_sphere_proportional_to_cube_travel_distance(
+    mut transformabels: QuerySet<(
```
Comment (alice-i-cecile) [inline examples/transforms/transform.rs:43]: Typo in variable name here: should be `transformables`

File: examples/transforms/transform.rs [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+        Query<(&Transform, &CubeState)>,
+        Query<(&mut Transform, &Center)>,
+    )>,
+) {
+    let distance: f32 = transformabels
+        .q0()
+        .iter()
+        .map(|(transform, cube)| (cube.start_pos - transform.translation).length())
```
Comment (alice-i-cecile) [inline examples/transforms/transform.rs:51]: This deserves an explanation for users that aren't familiar with functional programming.

Normally I prefer for loops in examples for that reason, but the .fold is much nicer this way.

**Comment by alice-i-cecile** [comment]
Good stuff! This is really helpful.

Two areas of content that I think should be addressed in this section eventually:

- more guidance on how to work with rotations: probably a few more examples for them and a pointer to [`Quat`]'s methods. I've seen lots of confusion around quaternions, axes of rotation, how to use them in a 2D context and so on. @bitshifter, would a docs PR (from me) to `glam`'s `Quat` with a glossary of terms be welcome here? We're just re-exporting it, so that's where the docs should be changed
- non-uniform scaling (@cart, have you come to a decision about what we should do about this in the engine as a whole?)

File: examples/transforms/transform.rs [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+        let incremental_turn_weight = cube.turn_speed * timer.delta_seconds();
+        transform.rotation = transform.rotation.lerp(full_turn, incremental_turn_weight);
+    }
+}
+
+fn scale_sphere_proportional_to_cube_travel_distance(
+    mut transformabels: QuerySet<(
+        Query<(&Transform, &CubeState)>,
+        Query<(&mut Transform, &Center)>,
+    )>,
```
Comment (MinerSebas) [inline examples/transforms/transform.rs:46]: You dont need to use a QuerySet, if you can prove that the queries are disjoint.
```suggestion
    cube: Query<(&Transform, &CubeState), Without<Center>>,
    center: Query<(&mut Transform, &Center)>,
    )>,
```

Comment (alice-i-cecile) [reply examples/transforms/transform.rs:46]: That's much clearer!

File: examples/transforms/scale.rs [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+        } else if transform.scale.length() < Vec3::ONE.length() {
+            // switch to next axis and expand
+            transform.scale = Vec3::ONE;
+            cube.scale_axis *= -1.0;
+            cube.scale_axis = Vec3::from((cube.scale_axis.z, cube.scale_axis.x, cube.scale_axis.y));
```
Comment (bitshifter) [inline examples/transforms/scale.rs:23]: Glam supports swizzling if you `use glam::swizzles::*` (or maybe `use bevy_math::swizzles::*`?) you should be able to do `cube.scale_axis.zxy()` here.

File: examples/transforms/transform.rs [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+) {
+    let distance: f32 = transformabels
+        .q0()
+        .iter()
+        .map(|(transform, cube)| (cube.start_pos - transform.translation).length())
+        .fold(0.0, |x, y| x + y);
+    for (mut transform, center) in transformabels.q1_mut().iter_mut() {
+        let new_size = Vec3::ONE - 0.05 * Vec3::ONE * distance;
+        let min_size = center.min_size * Vec3::ONE;
+        transform.scale = if new_size.length() < min_size.length() {
```
Comment (bitshifter) [inline examples/transforms/transform.rs:56]: If you don't need the result of `length` and are only performing a comparison then `length_squared` is much cheaper as it avoids a `sqrt`.

**Comment by bitshifter** [comment]
>     * more guidance on how to work with rotations: probably a few more examples for them and a pointer to [`Quat`]'s methods. I've seen lots of confusion around quaternions, axes of rotation, how to use them in a 2D context and so on. @bitshifter, would a docs PR (from me) to `glam`'s `Quat` with a glossary of terms be welcome here? We're just re-exporting it, so that's where the docs should be changed

I'm happy to look at a PR or an issue to improve documentation. What did you have in mind glossary wise? The use of quaternions for rotations in 2D may be a bit bevy specific? I'd not sure what docs for a non bevy audience would look like there. Probably the easiest way to progress would be via either https://github.com/bitshifter/glam-rs/discussions or https://github.com/bitshifter/glam-rs/issues.

File: examples/transforms/transform.rs (hunk 8 above)
Comment (NathanSWard) [reply examples/transforms/transform.rs:51]: I would actually disagree haha. 
I say use iterators all the way as that's the most idomatic rust code :wink:
And often it's clearer imo, since the iterator has a name where as a loop does not.

File: examples/transforms/transform.rs (hunk 9 above)
Comment (NathanSWard) [reply examples/transforms/transform.rs:46]: I'm curious, is there a specific performance gain either way?

Comment (alice-i-cecile) [reply examples/transforms/transform.rs:46]: I would be astonished if it's measurable.

File: examples/transforms/transform.rs (hunk 8 above)
Comment (alice-i-cecile) [reply examples/transforms/transform.rs:51]: Many of our users are coming from web / Python / C though, or are new to programming entirely ;) I tend to find beginners often get distracted from the point of the example when they hit iterator methods.

Also Cart agrees with me on that style point ;)

File: examples/transforms/rotation.rs (hunk 3 above)
Comment (mockersf) [reply examples/transforms/rotation.rs:5]: As most reference materials and tutorials for game and everything uses π, it's probably better to continue using it here instead of τ

File: examples/transforms/global_vs_local_translation.rs [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+        .add_system(stop_too_far_local_distance.system())
+        .add_system(stop_too_far_global_distance.system())
+        .run();
+}
+
+fn stop_too_far_global_distance(
+    mut cubes: Query<(&GlobalTransform, &mut SomeEntity, &GlobalStop)>,
+) {
+    for (transform, mut cube, _) in cubes.iter_mut() {
+        if (transform.translation - cube.spawn).length() >= MAX_DISTANCE {
```
Comment (mockersf) [inline examples/transforms/global_vs_local_translation.rs:35]:
```suggestion
    for (global_transform, mut cube, _) in cubes.iter_mut() {
        if (global_transform.translation - cube.spawn).length() >= MAX_DISTANCE {
```

File: examples/transforms/global_vs_local_translation.rs [hunk 13]
```diff
@@ -0,0 +1,116 @@
+use bevy::prelude::*;
+
+struct SomeEntity {
```
Comment (mockersf) [inline examples/transforms/global_vs_local_translation.rs:3]: I would prefer another name for this struct but I don't have any good idea for now...

File: examples/transforms/global_vs_local_translation.rs [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+
+    let main_entity_spawn: Transform = Transform::from_translation(Vec3::ZERO);
+    let subcomponent_transform = Transform::from_translation(Vec3::Y);
+    commands
+        .spawn_bundle(PbrBundle {
+            mesh: meshes.add(Mesh::from(shape::Cube { size: 1.0 })),
+            material: materials.add(StandardMaterial {
+                base_color: Color::YELLOW,
+                ..Default::default()
+            }),
```
Comment (mockersf) [inline examples/transforms/global_vs_local_translation.rs:84]:
```suggestion
            material: materials.add(Color::YELLOW.into()),
```

Comment (mockersf) [reply examples/transforms/global_vs_local_translation.rs:84]: same for the others colors

File: examples/transforms/transform.rs (hunk 8 above)
Comment (NathanSWard) [reply examples/transforms/transform.rs:51]: I guess for examples its ok.... 😝
Though for everything else iterators all the way... 🥳

**Comment by l4desu-mizu** [comment]
Just so you all know, I've seen all your comments and am baffled at the speed of your reaction. 😄
I'll work through your suggestions, but I might need some time. Hope you don't mind if i'm a bit slower with the rework.

File: examples/transforms/transform.rs (hunk 8 above)
Comment (l4desu-mizu) [reply examples/transforms/transform.rs:51]: I like to use the more functional approach, if the language has proper support for it.
But I think it is a valid suggestion, for anyone starting to learn a new Engine/Framework or even Language it is easier to concentrate on the specific aspects and not get confused by piled-up concepts from other realms.

**Comment by alice-i-cecile** [comment]
> Just so you all know, I've seen all your comments and am baffled at the speed of your reaction. 😄

Discord gives notifications when issues and PRs are made :) I also try to be a bit more responsive and involved for new contributors, since contributing to a new project can always be a bit intimidating.

> Hope you don't mind if i'm a bit slow with the rework.

Not at all; I don't think this is going to be blocking anything, as nice as it would be to have :)

File: examples/transforms/transform.rs (hunk 9 above)
Comment (l4desu-mizu) [reply examples/transforms/transform.rs:46]: This helped a lot, thanks!

File: examples/transforms/transform.rs (hunk 11 above)
Comment (l4desu-mizu) [reply examples/transforms/transform.rs:56]: I think I got it even simpler by just using the scalar values, since I was uniformly scaling, and added that bit just to prevent overly small (or negative sized) spheres. We'll see once I've pushed the update, I guess...

File: examples/transforms/global_vs_local_translation.rs (hunk 13 above)
Comment (l4desu-mizu) [reply examples/transforms/global_vs_local_translation.rs:3]: I'm going with EntityState for now since I use this struct mainly for state information.

**Comment by l4desu-mizu** [comment]
I hope I got everything and did not escalate too much.
I assume not everyone can trigger bors?

File: examples/transforms/global_vs_local_translation.rs [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+    // Spawn a basic cube to have an entity as reference.
+    let mut main_entity = commands.spawn_bundle(PbrBundle {
+        mesh: meshes.add(Mesh::from(shape::Cube { size: 1.0 })),
+        material: materials.add(Color::YELLOW.into()),
+        transform: main_entity_spawn,
+        ..Default::default()
+    });
+    main_entity
+        .insert(GlobalStop)
+        .insert(EntityState::new(main_entity_spawn.translation));
```
Comment (MinerSebas) [inline examples/transforms/global_vs_local_translation.rs:60]:
```suggestion
    let mut main_entity = commands.spawn_bundle(PbrBundle {
        mesh: meshes.add(Mesh::from(shape::Cube { size: 1.0 })),
        material: materials.add(Color::YELLOW.into()),
        transform: main_entity_spawn,
        ..Default::default()
    })
        .insert(GlobalStop)
        .insert(EntityState::new(main_entity_spawn.translation));
```
This will need a run of `cargo fmt` afterwards.

File: examples/transforms/transform.rs [hunk 16]
```diff
@@ -0,0 +1,148 @@
...
+        })),
+        material: materials.add(Color::YELLOW.into()),
+        transform: Transform::from_translation(Vec3::ZERO),
+        ..Default::default()
+    };
+    commands.spawn_bundle(center_sphere_mesh).insert(Center {
+        max_size: 1.0,
+        min_size: 0.1,
+        scale_factor: 0.05,
+    });
```
Comment (MinerSebas) [inline examples/transforms/transform.rs:50]: The extra variable is unnecessary, as it isn't used anywhere else.
```suggestion
    // Add an object (sphere) for visualizing scaling.
    commands
        .spawn_bundle(PbrBundle {
            mesh: meshes.add(Mesh::from(shape::Icosphere {
                radius: 3.0,
                subdivisions: 32,
            })),
            material: materials.add(Color::YELLOW.into()),
            transform: Transform::from_translation(Vec3::ZERO),
            ..Default::default()
        })
        .insert(Center {
            max_size: 1.0,
            min_size: 0.1,
            scale_factor: 0.05,
        });
```

File: examples/transforms/transform.rs [hunk 17]
```diff
@@ -0,0 +1,148 @@
...
+        mesh: meshes.add(Mesh::from(shape::Cube { size: 1.0 })),
+        material: materials.add(Color::WHITE.into()),
+        transform: cube_spawn,
+        ..Default::default()
+    };
+    commands.spawn_bundle(orbiting_cube_mesh).insert(CubeState {
+        start_pos: cube_spawn.translation,
+        move_speed: 2.0,
+        turn_speed: 0.2,
+    });
```
Comment (MinerSebas) [inline examples/transforms/transform.rs:71]:
```suggestion
    commands
        .spawn_bundle(PbrBundle {
            mesh: meshes.add(Mesh::from(shape::Cube { size: 1.0 })),
            material: materials.add(Color::WHITE.into()),
            transform: cube_spawn,
            ..Default::default()
        })
        .insert(CubeState {
            start_pos: cube_spawn.translation,
            move_speed: 2.0,
            turn_speed: 0.2,
        });
```

**Review by MinerSebas — APPROVED** [review]

File: examples/transforms/global_vs_local_translation.rs (hunk 15 above)
Comment (l4desu-mizu) [reply examples/transforms/global_vs_local_translation.rs:60]: I initially thought so too but due to the usage for spawning children afterwards the compiler won't let us do this.
error[E0716]: temporary value dropped while borrowed
  --> examples/transforms/global_vs_local_translation.rs:52:23
[...]
79 |       main_entity.with_children(|child_builder| {
   |       ----------- borrow later used here
   |
   = note: consider using a `let` binding to create a longer lived value

File: examples/transforms/transform.rs (hunk 16 above)
Comment (l4desu-mizu) [reply examples/transforms/transform.rs:50]: I've nothing in particular against your suggestion, but I actually purposefully moved the bundle creation out of the statement to have more but less convoluted statements. What would be preferred for examples?

File: examples/transforms/global_vs_local_translation.rs (hunk 15 above)
Comment (MinerSebas) [reply examples/transforms/global_vs_local_translation.rs:60]: Oh right, in that case it is fine as is.
You could do, but i dont know if that is better:
```rust
    let mut main_entity = commands.spawn();
    main_entity
        .insert_bundle(PbrBundle {
            mesh: meshes.add(Mesh::from(shape::Cube { size: 1.0 })),
            material: materials.add(Color::YELLOW.into()),
            transform: main_entity_spawn,
            ..Default::default()
        })
        .insert(GlobalStop)
        .insert(EntityState::new(main_entity_spawn.translation));
```

File: examples/transforms/transform.rs (hunk 16 above)
Comment (MinerSebas) [reply examples/transforms/transform.rs:50]: Currently, practically all examples construct the bundles inside the command calls.
The exceptions do this if a particular value is more difficult to determine.

File: examples/transforms/global_vs_local_translation.rs (hunk 15 above)
Comment (l4desu-mizu) [reply examples/transforms/global_vs_local_translation.rs:60]: I'd do it. keeps the structure similar to other examples.

**Comment by l4desu-mizu** [comment]
Hey,

I not sure how to further progress this.
So I've to bother you a little:
Do I have to do something? Am I missing something?

**Comment by alice-i-cecile** [comment]
No worries, thanks for checking in @l4desu-mizu. So the process from here is:

1. You comment in #2358 to verify that you're okay with this contribution being Apache + MIT licensed (rather than just MIT like Bevy was at the time of this PR).
2.  At least two other community members review and eventually approve the PR. I'm going to do this now, and see if I can wrangle up some more eyes on Discord.
3. We apply the `ready-for-final-review-label` to this PR.
4. Before launching 0.6, Cart and/or Francois will do a pass on all ready PRs and merge them in.

**Comment by alice-i-cecile** [comment]
One other thing @l4desu-mizu; can you mark the comment threads that you've addressed as resolved? That will make it much easier to review and verify that there aren't any outstanding issues.

File: examples/README.md [hunk 18]
```diff
@@ -231,6 +232,16 @@ Example | File | Description
...
 `bevymark` | [`tools/bevymark.rs`](./tools/bevymark.rs) | A heavy workload to benchmark your system with Bevy
 
+## Transforms
+
+Example | File | Description
+--- | --- | ---
+`global_vs_local_translation` | [`transforms/global_vs_local_translation.rs`](./transforms/global_vs_local_translation.rs) | Illustrates the difference between direction of a translation in respect to local object or global object Transform
+`rotation` | [`transforms/rotation.rs`](./transforms/rotation.rs) | Illustrates how to (constantly) rotate an object around an axis
+`scale` | [`transfroms/scalse.rs`](./transforms/scale.rs) | Illustrates how to scale an object in each direction
+`transform` | [`transforms/transfrom.rs`](./transfroms/transform.rs) | Shows multiple transformations of objects
```
Comment (alice-i-cecile) [inline examples/README.md:242]:
```suggestion
`transform` | [`transforms/transfrom.rs`](./transforms/transform.rs) | Shows multiple transformations of objects
```

File: examples/README.md [hunk 19]
```diff
@@ -231,6 +232,16 @@ Example | File | Description
 --- | --- | ---
 `bevymark` | [`tools/bevymark.rs`](./tools/bevymark.rs) | A heavy workload to benchmark your system with Bevy
 
+## Transforms
+
+Example | File | Description
+--- | --- | ---
+`global_vs_local_translation` | [`transforms/global_vs_local_translation.rs`](./transforms/global_vs_local_translation.rs) | Illustrates the difference between direction of a translation in respect to local object or global object Transform
+`rotation` | [`transforms/rotation.rs`](./transforms/rotation.rs) | Illustrates how to (constantly) rotate an object around an axis
+`scale` | [`transfroms/scalse.rs`](./transforms/scale.rs) | Illustrates how to scale an object in each direction
```
Comment (alice-i-cecile) [inline examples/README.md:241]:
```suggestion
`scale` | [`transforms/scalse.rs`](./transforms/scale.rs) | Illustrates how to scale an object in each direction
```

File: examples/transforms/global_vs_local_translation.rs [hunk 20]
```diff
@@ -0,0 +1,141 @@
...
+
+// Define a marker for entities that should stop at a distance in respect to their global transform.
+struct GlobalStop;
+// Define a marker for entities that should stop at a distance in respect to their local transform.
+struct LocalStop;
+
+// Define the maximum distance an entity should be able to move away from its spawn.
+const MAX_DISTANCE: f32 = 5.0;
+
+// Ease up creation of EntityState structs.
```
Comment (alice-i-cecile) [inline examples/transforms/global_vs_local_translation.rs:17]: I think this is probably overkill for this example, and would just use struct literal syntax to initialize these objects.

File: examples/transforms/global_vs_local_translation.rs [hunk 21]
```diff
@@ -0,0 +1,141 @@
+use bevy::prelude::*;
+
+// Define a struct to store basic state information.
+struct EntityState {
```
Comment (alice-i-cecile) [inline examples/transforms/global_vs_local_translation.rs:4]: I'm don't think this is the clearest way to store this information: generally, I'd pass in the spawn location to the entity spawning system, and then have a separate `Stopped` or `IsStopped(bool)` component.

File: examples/transforms/global_vs_local_translation.rs [hunk 22]
```diff
@@ -0,0 +1,141 @@
...
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    // To show the difference between a local transform (rotation, scale and position in respect to a given entity)
+    // and global transform (rotation, scale and position in respect to the base coordinate system of the visible scene)
+    // it's helpful to add multiple entities that are attached to each other.
+    // This way we'll see that the transform in respect to an entities parent is different to the
+    // global transform within the visible scene.
+    // This example focuses on translation only to make a simple case for the differences.
```
Comment (alice-i-cecile) [inline examples/transforms/global_vs_local_translation.rs:48]:
```suggestion
    // This example focuses on translation only to clearly demonstrate the differences.
```

**Review by alice-i-cecile — COMMENTED** [review]
Looks good! I left a few comments and typo fixes, but once those are cleaned up this looks good to me.

**Comment by RustyStriker** [comment]
I compiled the examples after reading them and these are my notes:

`rotation` was straight forward, a moving cube with a short file, not too much to read if you want to understand what you are looking at

`translation` pretty much gets the same notes as rotation, not a lot going on and the code is also quick to read through

`scale` is a bit weird since it scales in 1 axis every time, and the camera is facing towards the cube so it makes it a bit weird to look at sometimes(maybe rotate the cube quarter pi so we will look at its edge?)

the `global_vs_local_transform` example is really confusing, the code itself is also very long which makes it harder to understand(the green cube is inside the red cube for the first half of the movement).

`transform` has a lot going one, and the code is rather long, i dont think this example really manages to portray transform(tho i am not sure why you need both example for transform and each of its components)

Generally(applies to all), I think the examples can be interactive, even in a minimal way like changing the rotation axis in `rotation`, but it doesnt have to be for the simpler one.
for the `transform` and `global_vs_local_translation` I think interactive-ness will go a long way with allowing users to better understand it(or maybe some text explaining what im looking at and what the relations are)

File: examples/README.md (hunk 19 above)
Comment (Weibye) [inline examples/README.md:241]:
```suggestion
`scale` | [`transforms/scale.rs`](./transforms/scale.rs) | Illustrates how to scale an object in each direction
```
There are two typos in this :)

**Review by Weibye — CHANGES_REQUESTED** [review]
This looks good to me, just a typo in a link :)

**Comment by l4desu-mizu** [comment]
@RustyStriker 

> `scale` is a bit weird since it scales in 1 axis every time, and the camera is facing towards the cube so it makes it a bit weird to look at sometimes(maybe rotate the cube quarter pi so we will look at its edge?)

sure.

> the `global_vs_local_transform` example is really confusing, the code itself is also very long which makes it harder to understand(the green cube is inside the red cube for the first half of the movement).

This is honestly just me trying to get the difference between local/global transforms somewhat descriptive in code. But I admit I had similar thoughts, without really any better approaches. I'd draw coordinate systems but that also seemed a bit much.
However, I guess moving the cubes parallely - with some distance - would be a first step in the right direction?

> `transform` has a lot going one, and the code is rather long, i dont think this example really manages to portray transform(tho i am not sure why you need both example for transform and each of its components)

True. This was merely to capture all possible transformations in one example as the issue suggested. I just thought it would be better to transform multiple objects instead of one.

> Generally(applies to all), I think the examples can be interactive, even in a minimal way like changing the rotation axis in `rotation`, but it doesnt have to be for the simpler one.
> for the `transform` and `global_vs_local_translation` I think interactive-ness will go a long way with allowing users to better understand it(or maybe some text explaining what im looking at and what the relations are)

I haven't looked too much into inputs and refrained in doing so for these examples. I thought that would distract from the transformation code.
What would be your suggestion. Changing the rotation axis via key-press for example? Or even connecting the transform to mouse-movement?

File: examples/transforms/rotation.rs (hunk 3 above)
Comment (l4desu-mizu) [reply examples/transforms/rotation.rs:5]: I'm sticking with PI for now, due to mockersf suggestion.

**Comment by l4desu-mizu** [comment]
@alice-i-cecile 

>     1. You comment in [Bevy relicensing #2358](https://github.com/bevyengine/bevy/discussions/2358) to verify that you're okay with this contribution being Apache + MIT licensed (rather than just MIT like Bevy was at the time of this PR).

Just to verify I'm not mistaken. You linked the (i guess initial discussion) for the relicense. However, I've seen that there was an issue contributers were guided to, to state their agreement/approval for the relicense. So I went there and did the same https://github.com/bevyengine/bevy/issues/2373.

**Comment by alice-i-cecile** [comment]
> Just to verify I'm not mistaken. You linked the (i guess initial discussion) for the relicense. However, I've seen that there was an issue contributers were guided to, to state their agreement/approval for the relicense. So I went there and did the same #2373.

Ah yep, that's the correct place for it! That's what I get for relying on Github's suggestions. Thanks; I'll remove the label now :D

**Comment by RustyStriker** [comment]
@l4desu-mizu

> This is honestly just me trying to get the difference between local/global transforms somewhat descriptive in code. But I admit I had similar thoughts, without really any better approaches. I'd draw coordinate systems but that also seemed a bit much.
However, I guess moving the cubes parallely - with some distance - would be a first step in the right direction?

Indeed that is what came to my mind, just so you see all of them all the time.

> I haven't looked too much into inputs and refrained in doing so for these examples. I thought that would distract from the transformation code.
What would be your suggestion. Changing the rotation axis via key-press for example? Or even connecting the transform to mouse-movement?

I think something like a 2 buttons example is enough, so for movement you can do `move left/right`, it is minimal interactivity and should add something like ~8 lines to the relevant system(getting keyboard input is incredibly easy)

I am not sure how necessary it will be for the simpler examples, but for the more complex ones(`global_vs_local_translation`) it should have a much better explanation.

I also thought about the idea of adding a text explaining the point of the example, not in a complicated manner but more of if someone is sitting by your side and telling you what is actually happening, but I am really not sure how helpful it will be.

A big problem with these examples is that they showcase concepts more than a "how to" or a "This is how something might look in bevy" kind of example, so actually reading the code might not be too necessary(and if you show the relevant lines on the screen in a text dump the can even avoid it completely if they want).

(I really feel like im going too far at this)

File: examples/transforms/global_vs_local_translation.rs (hunk 21 above)
Comment (l4desu-mizu) [reply examples/transforms/global_vs_local_translation.rs:4]: I'd change it to Moving (instead of Stopped) and remove the component once it should stop. Also makes filtering easier since it's in the queries and is probably more "the bevy/ecs way"?
I assume that was on your mind?

File: examples/transforms/scale.rs [hunk 23]
```diff
@@ -41,7 +41,7 @@ fn setup(
         .spawn_bundle(PbrBundle {
             mesh: meshes.add(Mesh::from(shape::Cube { size: 1.0 })),
             material: materials.add(Color::WHITE.into()),
-            transform: Transform::from_translation(Vec3::ZERO),
+            transform: Transform::from_rotation(Quat::from_rotation_y(PI / 4.0)),
```
Comment (RustyStriker) [inline examples/transforms/scale.rs:44]: You didnt import pi here

Comment (RustyStriker) [inline examples/transforms/scale.rs:44]: You didn't import PI

**Review by RustyStriker — APPROVED** [review]
You should probably compile against the updated main branch and change the `App::build()` to `App::new()` so it will compile when a new user tries to run it  through main, but overall great job

File: examples/transforms/scale.rs (hunk 23 above)
Comment (l4desu-mizu) [reply examples/transforms/scale.rs:44]: If I'm correct bevy won't need the explicit .system() call in future? Should I remove those too? (with the merged main)

Comment (RustyStriker) [reply examples/transforms/scale.rs:44]: It seems they removed the `.system()` in the other examples, so yes indeed, nice catch :)

**Comment by l4desu-mizu** [comment]
So I tried to rework the global vs local translation example as suggested by RustyStriker. I do think the code does get somewhat simpler when enabling interaction even though it now contains multiple aspects of bevy. However it's a more complex thing and I hope the way I defined the systems keeps it somewhat distinguished.
For now I'd like to refrain from changing the more simpler examples for the same reason. It'd seem to me that adding inputs and even text would distract form the basic goal of the example, even if the current way for decisions in these examples might not be ideal...

However, I try to keep up to date with main, but I haven't figured out what the advisories-check wants me to do. I don't think I changed dependencies, so it's a bit confusing to me that it says that there are security advisories or unmaintained crates. I cannot see additional information either. Is someone of you able to help me with that?

**Comment by alice-i-cecile** [comment]
> However, I try to keep up to date with main, but I haven't figured out what the advisories-check wants me to do. I don't think I changed dependencies, so it's a bit confusing to me that it says that there are security advisories or unmaintained crates. I cannot see additional information either. Is someone of you able to help me with that?

This is a global Bevy issue: one of our dependencies started triggering this about a month back. Your PR is fine; just ignore it.

**Review by alice-i-cecile — APPROVED** [review]
